Modules/Devices/test/Devices.Application.Tests/Tests/Tiers/Queries/ListTiers/HandlerTests.cs

```diff
@@ -0,0 +1,94 @@
+﻿using Backbone.Modules.Devices.Application.Infrastructure.Persistence.Repository;
+using Backbone.Modules.Devices.Application.Tests;
+using Backbone.Modules.Devices.Application.Tiers.Queries.ListTiers;
+using Backbone.Modules.Devices.Domain.Aggregates.Tier;
+using Enmeshed.BuildingBlocks.Application.Abstractions.Infrastructure.Persistence.Database;
+using Enmeshed.BuildingBlocks.Application.Pagination;
+using FakeItEasy;
+using FluentAssertions;
+using FluentAssertions.Execution;
+using Xunit;
+using Handler = Backbone.Modules.Devices.Application.Tiers.Queries.ListTiers.Handler;
+
+namespace Backbone.Modules.Devices.Application.Tests.Tests.Tiers.Queries.ListTiers;
+
+public class HandlerTests
+{
+    private readonly ITierRepository _fakeRepository;
+    private readonly Handler _handler;
+
+    public HandlerTests()
+    {
+        AssertionScope.Current.FormattingOptions.MaxLines = 1000;
+
+        _fakeRepository = A.Fake<ITierRepository>();
+
+        _handler = CreateHandler(); 
+    }
+
+    [Fact]
+    public async void Returns_an_empty_list_when_no_tiers_exist()
+    {
+        // Arrange
+        var tiersList = new List<Tier>();
+        var request = new PaginationFilter() { PageSize = 5 };
+        A.CallTo(() => _fakeRepository.FindAll(request)).Returns(MakeDBPaginationResult(tiersList));
```

Comment:
You should create a `ListTiersStubRepository` that accepts a list of Tiers in the constructor. This makes the test a lot more readable:

```suggestion
        var stubRepository = new ListTiersStubRepository(MakeDBPaginationResult(tiersList));
```

Do the same for the other tests - and then remove the `_fakeReposistory` field